src/components/DeliverableView/DocumentsUploader.tsx

```diff
@@ -21,13 +25,44 @@ const DocumentsUploader = ({ deliverable }: ViewProps): JSX.Element => {
     }
   }, [activeLocale, deliverable.templateUrl]);
 
-  const onCloseFileUploadDialog = useCallback(() => void setFiles([]), []);
+  // determine if the status change confirmation dialog is needed
+  const showStatusChangeDialog = useMemo(() => {
+    // if the status change has already been confirmed, don't show the dialog
+    if (statusChangeConfirmed) {
+      return false;
+    }
+
+    // participant view
+    if (!isAcceleratorRoute && !['Not Submitted', 'In Review', 'Needs Translation'].includes(deliverable.status)) {
+      return true;
+    }
+
+    // accelerator view
+    if (isAcceleratorRoute && !['Not Submitted', 'In Review'].includes(deliverable.status)) {
+      return true;
+    }
```

Comment:
I think you can ignore the `isAcceleratorRoute`. According to Figma the condition is always checking for these 3 statuses. Can just `return ['Not Submitted', 'In Review', 'Needs Translation'].includes(deliverable.status);`

Reply:
@nickgraz "Needs Translation" should be for the participant view only according to Karthik's comment in the ticket for SW-4945.

Reply:
That is for display only, the underlying status of the deliverable will not change. In other words, a participant can still interact with a deliverable that has a status of "needs translation", they just won't know it because to them it will _appear_ "in review". Ultimately, this component should be view-agnostic anyways.

Reply:
For participants yes we hide the 'needs translation' semantic, it's still technically 'in review'.
For internal users, we show the 'needs translation' semantic, we need to let them know the status will change from 'needs translation' back to 'in review' on a new document upload.
Status changes are handled in the backend.
I can't find the figma thread that discussed this change, the original thread does not mention the change.

If we don't want the isAccelerator dependency here, we could pass the list of statuses to ignore as a prop.

Reply:
But the underlying status will still be "needs translation", even if we don't show it to participants, so the logic is the same, right? In other words, if a participant uploads a document to a deliverable that is "needs translation" (to them it looks like "in review"), the status is still reset and therefore we should still show them a dialog that the status is going to be reset. 

I suppose it will appear as though nothing has changed (on the participant side), since the "reset" status is also "in review". Maybe that is why you are saying it should not be shown? Since it doesn't explicitly call out that it will put it back into "in review" it still seems like useful information for the user that their upload is going to modify the deliverable's status, but if this has already been sorted out in Figma or something then I'm OK with having it do two different things. But I would prefer that the statuses are passed in and the component becomes view agnostic.

Reply:
Yes, that is correct - we don't need to say it will be reset because it already appears as 'In Review' to the participant (even if it is 'Translation Needed' in actual data). The BE will reset it from 'Translation Needed' to 'In Review' behind the scenes. There is some resetting going on, we just won't flag it to the participant user.
Passing these statuses into the component sounds good to me 👍